pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -131,60 +131,74 @@ public synchronized CompletableFuture<Void> addConsumer(Consumer consumer) {
+                consumer.setPendingAcksRemoveHandler(new PendingAcksMap.PendingAcksRemoveHandler() {
+                    @Override
+                    public void handleRemoving(Consumer consumer, long ledgerId, long entryId, int stickyKeyHash,
+                                               boolean closing) {
+                        drainingHashesTracker.reduceRefCount(consumer, stickyKeyHash, closing);
                     }
-                    return result;
-                })
-        ).thenRun(() -> {
-            synchronized (PersistentStickyKeyDispatcherMultipleConsumers.this) {
-                if (recentlyJoinedConsumerTrackingRequired) {
-                    final Position lastSentPositionWhenJoining = updateIfNeededAndGetLastSentPosition();
-                    if (lastSentPositionWhenJoining != null) {
-                        consumer.setLastSentPositionWhenJoining(lastSentPositionWhenJoining);
-                        // If this was the 1st consumer, or if all the messages are already acked, then we
-                        // don't need to do anything special
-                        if (recentlyJoinedConsumers != null
-                                && consumerList.size() > 1
-                                && cursor.getNumberOfEntriesSinceFirstNotAckedMessage() > 1) {
-                            recentlyJoinedConsumers.put(consumer, lastSentPositionWhenJoining);
-                        }
+
+                    @Override
+                    public void startBatch() {
+                        drainingHashesTracker.startBatch();
                     }
-                }
+
+                    @Override
+                    public void endBatch() {
+                        drainingHashesTracker.endBatch();
+                    }
+                });
+                registerDrainingHashes(consumer, impactedConsumers);
+            }
+        }).exceptionally(ex -> {
+            internalRemoveConsumer(consumer);
+            throw FutureUtil.wrapToCompletionException(ex);
+        });
+    }
+
+    private synchronized void registerDrainingHashes(Consumer skipConsumer,
+                                                     ImpactedConsumersResult impactedConsumers) {
+        impactedConsumers.processRemovedHashRanges((c, removedHashRanges) -> {
+            if (c != skipConsumer) {
+                c.getPendingAcks().forEach((ledgerId, entryId, batchSize, stickyKeyHash) -> {
+                    if (stickyKeyHash == STICKY_KEY_HASH_NOT_SET) {
+                        log.warn("[{}] Sticky key hash was missing for {}:{}", getName(), ledgerId, entryId);
+                        return;
+                    }
+                    if (removedHashRanges.containsStickyKey(stickyKeyHash)) {
+                        // add the pending ack to the draining hashes tracker if the hash is in the range
+                        drainingHashesTracker.addEntry(c, stickyKeyHash);
```

Comment:
**Background**
- `drainingHashesTracker.addEntry(c, stickyKeyHash)` will increase the variable `DrainingHashEntry.refCount`.  

**Issue**: the variable named `DrainingHashEntry.refCount` will be increased repeatedly at the following scenario:
- Register `C1`, assign `k1` to `C1`
- Register `C2`, move `k1` to `C2`
  - `DrainingHashEntry.refCount` increases 
- Register `C3`, move `k1` to `C3`
  - `DrainingHashEntry.refCount` increases

Reply:
Resolved as a non-issue.